dask/array/core.py

```diff
@@ -4025,6 +4025,11 @@ def broadcast_arrays(*args, **kwargs):
     if kwargs:
         raise TypeError("unsupported keyword argument(s) provided")
 
+    # Unify uneven chunking
+    inds = [list(range(x.ndim))[::-1] for x in args]
+    uc_args = list(concat(zip(args, inds)))
```

Comment:
No need for the `list` call here, the splat below iterates automatically.

```suggestion
    uc_args = concat(zip(args, inds))
```